src/utils/editor/source-search.js

```diff
@@ -157,6 +157,31 @@ function doSearch(ctx, rev, query, keepSelection, modifiers: SearchModifiers) {
   });
 }
 
+export function highlightQuery(
+  ctx: any,
+  rev: boolean,
+  query: string,
+  keepSelection: boolean,
+  modifiers: SearchModifiers,
+  line: number
+) {
+  const { cm } = ctx;
+  if (!cm) {
+    return;
+  }
+  const defaultIndex = { line: -1, ch: -1 };
+
+  return cm.operation(function() {
+    const state = getSearchState(cm, query);
+    const isNewQuery = state.query !== query;
+    state.query = query;
+
+    updateOverlay(cm, state, query, modifiers);
+    updateCursor(cm, state, keepSelection);
+    findNextOnLine(ctx, rev, query, isNewQuery, modifiers, line);
+  });
+}
+
```

Comment:
do we want to add test coverage here:
https://github.com/AlexisDeschamps/debugger.html/blob/3829-search-on-match/src/utils/editor/tests/source-search.spec.js#L1

Reply:
Yes, I will investigate further for the tests.